### ISSUE TITLE
Support for exporting Redshift cluster nodes and/or IP addresses

### ISSUE THREAD

**Issue by ziggythehamster** [hashicorp/terraform-provider-aws#20727]
*(this is a resubmission of #3394)*

Currently aws_redshift_cluster resources do not support exporting the cluster nodes. However it is already possible to get those back from AWS using the CLI. The cluster nodes are useful for determining the public and private IP addressed of the nodes.

While attaching an EIP seems like a possible workaround, it is quite limited when the cluster is not publicly accessible and deployed in a private subnet (which is the recommended way).

Here's a small example for to do that using the AWS CLI:

```sh
aws redshift describe-clusters --cluster-identifier <cluster-identifier>
```

https://docs.aws.amazon.com/redshift/latest/dg/load-from-host-steps-retrieve-key-and-ips.html

**Adding to the original request:**

I would also like to add that being able to filter the list by node role is a requirement for me - I currently use an external datasource that does this, so I can get the leader node private IP:

```sh
aws --region $AWS_REGION redshift describe-clusters --cluster-identifier $REDSHIFT_CLUSTER_ID | jq -c '.Clusters[0].ClusterNodes | map(select(.NodeRole == "LEADER"))[0] | { private_ip_address: .PrivateIPAddress, public_ip_address: .PublicIPAddress }'
```

# Terraform Version

Any

# Affected Resource(s)

* aws_redshift_cluster (and the data source)

**Comment by ziggythehamster** [comment]
Someone had started a PR at #4563, as well.

**Comment by ziggythehamster** [comment]
#4563 got merged into 3.57.0, which satisfies this feature request (I can use a for in loop with a filter to find the leader node easily). I'm going to close this. Thanks!
